.github/workflows/CI.yml

```diff
@@ -45,11 +45,6 @@ jobs:
       with:
         java-version: ${{ matrix.java }}
         distribution: temurin
-    - name: Build and Run Tests
+    - name: Build
       run: |
-          ./gradlew check
-    - name: Upload Coverage Report
-      if: matrix.os == 'ubuntu-latest'
-      uses: codecov/codecov-action@v3
-      with:
-        file: ./build/reports/jacoco/test/jacocoTestReport.xml
+          ./gradlew check -x test -x integTest -x yamlRestTest
```

Comment:
just wondering, why not use gradlew build here?

Reply:
We can. I don't have any strong opinion here.

Reply:
I am good with either, `./gradlew build` means we also run assemble which runs the `:compileJava, :processResources` which `check` does also but this does it into a jar.

Reply:
Changed it to build.

Reply:
I think we should step back and figure out all the dependencies and figure out what we are trying to optimize here.  With the current PR we're shifting the tests somewhere else (still in series so not really gaining much) but duplicating compile/build stuff in parallel.

My general preference would be for CI to keep all the check stuff and `test` and code coverage bits, and just spin off the integ tests and yaml tests into another job in parallel (or just put them in the CI yaml at the top level under "jobs").

Lots of different ways to do it, haven't really been convinced we have a problem that needs solving yet :)

Reply:
I agree with you. I will go ahead and close the PR. We can always come back and open it back when the plugin becomes heavy.